### PR TITLE
Transformations: Fields to labels 

### PR DESCRIPTION
**Pull request by douglaswth**
**What this PR does / why we need it**:

While there is a "Labels to fields" transformation, there is currently no way to do the reverse. This would be especially useful when using mixed data sources along with "Labels to fields" in order to merge them together and make calculations where you want to display your results as a graph with multiple lines or some other similar visualization.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ryantxu** [comment]
Hymm -- when limited to timeseries, maybe this is the same as "wide to long" that we currently support in the backend :thinking: 
https://github.com/grafana/grafana-plugin-sdk-go/blob/main/data/time_series.go#L381

cc @kylebrandt

**Comment by ryantxu** [comment]
Rather than adding a new transformation, should this implement "wide to long" here:
https://github.com/grafana/grafana/blob/main/public/app/core/components/TransformersUI/prepareTimeSeries/prepareTimeSeries.ts#L24

**Comment by ryantxu** [comment]
@douglaswth should we move this to the "prepare timeseries" transformer -- so that does "wide to long"?

**Comment by douglaswth** [comment]
@ryantxu Okay, I'll take a stab at that after I finish up the other PR.

File: docs/sources/panels/transformations/fields-to-labels.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
++++
+title = "Fields to labels"
+weight = 300
++++
+
+# Fields to labels transform
+
+> **Note:** This is a new beta transformation introduced in v8.1.
+
+This transformation allows you to split time series to multiple time series based on the values of selected fields which also become label values for their respective time series. This transformation provides the inverse behavior to the [Labels to fields]({{< relref "./types-options.md#labels-to-fields" >}}).
```
Comment (achatterjee-grafana) [inline docs/sources/panels/transformations/fields-to-labels.md:10]:
```suggestion
This transformation allows you to split time series data into multiple time series based on the values of selected fields. These selected fields become label values for their respective time series. Selecting multiple **Label fields** as label value results in as many time series as the unique label value combinations in your input data.
This behavior of this transformation is the opposite of [Labels to fields]({{< relref "./types-options.md#labels-to-fields" >}}) transform.
```

File: docs/sources/panels/transformations/fields-to-labels.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  | 2021-08-17 2:15:05 | 1 | 3 |
+- Series 2: label Datacenter=US
+  | Time | Server A | Server B |
+  | ------------------ | -------- | -------- |
+  | 2021-08-17 2:14:35 | 3 | 4 |
+  | 2021-08-17 2:15:05 | 3 | 5 |
+
+## Configuration
+
+You may select multiple **Label fields** which will be used as label values. As a result you will get as many time series as there are unique label field combinations coming from your input data.
```
Comment (achatterjee-grafana) [inline docs/sources/panels/transformations/fields-to-labels.md:38]: remove

File: docs/sources/panels/transformations/fields-to-labels.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
++++
+title = "Fields to labels"
```
Comment (achatterjee-grafana) [inline docs/sources/panels/transformations/fields-to-labels.md:2]: Do we need a new topic here? Let's add this information in the "Transformation types and options" topic, as a section after https://grafana.com/docs/grafana/latest/panels/transformations/types-options/#labels-to-fields.

File: docs/sources/panels/transformations/fields-to-labels.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# Fields to labels transform
+
+> **Note:** This is a new beta transformation introduced in v8.1.
+
+This transformation allows you to split time series to multiple time series based on the values of selected fields which also become label values for their respective time series. This transformation provides the inverse behavior to the [Labels to fields]({{< relref "./types-options.md#labels-to-fields" >}}).
+
+## Example
+
+Given a query result with a single time series:
```
Comment (achatterjee-grafana) [inline docs/sources/panels/transformations/fields-to-labels.md:14]:
```suggestion
The following table shows the query result with a single time series:
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added doc review.

**Comment by ryantxu** [comment]
@douglaswth should we close this in favor of https://github.com/grafana/grafana/pull/38670

**Comment by douglaswth** [comment]
Closing in favor of #38670.

**Pull request closed without merging**